src/containers/MyNdla/Folders/FolderSharedPage.tsx

```diff
@@ -0,0 +1,222 @@
+import { routes } from "../../../routeHelpers";
+import ErrorPage from "../../ErrorPage";
+import NotFound from "../../NotFoundPage/NotFoundPage";
+import SharedFolder from "../../SharedFolderPage/SharedFolder";
+import MyNdlaPageWrapper from "../components/MyNdlaPageWrapper";
+import SettingsMenu, { MenuItemProps } from "../components/SettingsMenu";
+import { useFolders, useGetSharedFolder, useSaveFolderResourceMutation } from "../folderMutations";
+
+const flattenResources = (folder: GQLFolder): GQLFolderResource[] => {
+  const subResources = folder.subfolders.flatMap(flattenResources);
+  return folder.resources.concat(subResources);
+};
+
+const FolderSharedPage = () => {
+  const [showSaveLinkButton, setShowSaveLinkButton] = useState(false);
+  const { folderId = "" } = useParams();
+  const { authenticated } = useContext(AuthContext);
+  const [viewType, setViewType] = useState<ViewType>("list");
+
+  const { loading: foldersLoading, sharedFolders } = useFolders();
+
+  const {
+    folder,
+    loading: sharedFolderLoading,
+    error,
+  } = useGetSharedFolder({
+    id: folderId,
+    includeResources: true,
+    includeSubfolders: true,
+  });
+
+  const loading = foldersLoading || sharedFolderLoading;
+
+  const buttons = useMemo(() => {
+    if (!folder) return [];
+    const copyFolder = (
+      <CopyFolderModal folder={folder}>
+        <ButtonV2 aria-label={t("myNdla.folder.copy")} variant="ghost">
+          <Copy />
+          {t("myNdla.folder.copy")}
+        </ButtonV2>
+      </CopyFolderModal>
+    );
+
+    const saveLink = <SaveLink setShowTrigger={() => setShowSaveLinkButton(false)} folder={folder} />;
+    const actions = [copyFolder];
+
+    if (showSaveLinkButton) {
+      actions.push(saveLink);
+    }
+
+    return actions;
+  }, [folder, showSaveLinkButton]);
+
+  const menu = useMemo(() => {
+    if (!folder) return [];
+    const copyFolder: MenuItemProps = {
+      text: t("myNdla.folder.copy"),
+      isModal: true,
+      icon: <Copy />,
+      modalContent: (close) => <CopyFolder folder={folder} onClose={close} />,
+    };
+
+    const saveLink: MenuItemProps = {
+      text: t("myNdla.folder.sharing.button.saveLink"),
+      isModal: true,
+      icon: <Subject />,
+      modalContent: (close) => (
+        <SaveLinkContent setShowTrigger={() => setShowSaveLinkButton(false)} folder={folder} onClose={close} />
+      ),
+    };
+
+    const actions = [copyFolder];
+
+    if (showSaveLinkButton) {
+      actions.push(saveLink);
+    }
+
+    return <SettingsMenu menuItems={actions} />;
+  }, [folder, showSaveLinkButton]);
+
+  useEffect(() => {
+    if (!sharedFolders) return;
+    setShowSaveLinkButton(!sharedFolders?.some((f) => f.id === folderId));
+  }, [folderId, sharedFolders]);
+
+  if (!authenticated) {
+    return <Navigate to={routes.folder(folderId)} />;
+  }
+
+  if (loading) {
+    return <Spinner />;
+  } else if (error?.graphQLErrors[0]?.extensions?.status === 404) {
```

Comment:
Hvis vi på sikt tenker å ta i bruk biome i dette prosjektet også så kan jo dette endres med en gang. Hva tenker du @Jonas-C ? 
Ref: [#2243](https://github.com/NDLANO/frontend-packages/pull/2243)

Reply:
Ikke kjempefarlig for meg, men forsåvidt!